pgxn/neon/pagestore_smgr.c

```diff
@@ -2425,6 +2425,19 @@ neon_exists(SMgrRelation reln, ForkNumber forkNum)
 			.forknum = forkNum
 		};
 
+		/*
+		 * We may get to this line while aborting the transaction if pageserger
+		 * can't be reached. If we create another pageserver request, it will abort,
+		 * causing an abort of already aborting transaction, which in turn yields
+		 * a sigabrt due to failed Assert.
+		 * Ideally, we'd just call IsAbortedTransactionBlockState but the issue is
+		 * transaction's state is TRANS_ABORT but blockState is TBLOCK_INPROCESS so
+		 * it doesn't work. Therefore we do this hack: function will return false
+		 * if current transaction is in TRANS_ABORT state.
+		 * Note: it depends on implementation of SubTransactionIsActive
```

Comment:
Hmm, so this is a way to basically do `CurrentTransactionState->state == TRANS_ABORT`, which we can't do directly here because CurrentTransactionState is `static`. Clever trick :-).